### PR TITLE
Implemented UG documentation for treeview appearance and scrolling

### PR DESCRIPTION
**Pull request by Dhivya-SF4094**
Added TreeView appearance and scrolling UG documentation

### CODE REVIEW

File: MAUI/TreeView/scrolling.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+{
+    var count= viewModel.ImageNodeInfo.Count;
+    var data = viewModel.ImageNodeInfo[count-1];
+    TreeView.BringIntoView(data, false, true);
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+N> We need to set the `NodePopulationMode` API value as `PopulationMode.Instant` for scrolling to the collapsed item in addition to the additional parameter passed to the `BringIntoView` method.
```
Comment (florenceliliana) [inline MAUI/TreeView/scrolling.md:75]: N> We need to set the `NodePopulationMode` API value as a `PopulationMode.Instant` for scrolling the collapsed item in addition to the additional parameter passed to the `BringIntoView` method.

**Review by florenceliliana — CHANGES_REQUESTED** [review]
I have completed the review.

File: MAUI/TreeView/appearance.md [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+title: Appearance | TreeView for Maui | Syncfusion
+description: Learn about Appearance support in Syncfusion Maui Treeview (SfTreeView) Control and more details.
+platform: MAUI
+control: SfTreeView
+documentation: ug
+---
+
+# Appearance in Maui TreeView (SfTreeView)
+
+The .NET MAUI SfTreeView allows customizing appearance of the underlying data, and provides different functionalities to the end-user.
```
[the remark this answers is not in the document]
Comment (Dhivya-SF4094) [reply MAUI/TreeView/appearance.md]: Replaced the content as mentioned above

File: MAUI/TreeView/appearance.md [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+        }
+    }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+## BindingContext for ItemTemplate
+
+By default, the binding context of tree view item will be the data model object for Bound Mode and `TreeViewNode` for Unbound Mode.
```
[the remark this answers is not in the document]
Comment (Dhivya-SF4094) [reply MAUI/TreeView/appearance.md:79]: Replaced the content as mentioned above.

File: MAUI/TreeView/appearance.md [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+                return grid;
+            });
+            this.Content = treeView;
+        }
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+Similarly, you can customize the expander view by using `ExpanderTemplate` property like above example.
```
[the remark this answers is not in the document]
Comment (Dhivya-SF4094) [reply MAUI/TreeView/appearance.md:141]: Replaced the content as mentioned above

File: MAUI/TreeView/appearance.md [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+
+# Appearance in Maui TreeView (SfTreeView)
+
+The .NET MAUI SfTreeView allows customizing appearance of the underlying data, and provides different functionalities to the end-user.
+
+## ItemTemplate
+ 
+A template can be used to present the data in a way that makes sense for the application by using different controls.
+
+The TreeView allows you to customize the appearance of content view and expander view by setting the `ItemTemplate` and `ExpanderTemplate` properties.
```
[the remark this answers is not in the document]
Comment (Dhivya-SF4094) [reply MAUI/TreeView/appearance.md:18]: Replaced the content as mentioned above.

File: MAUI/TreeView/appearance.md [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+        }
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+Similarly, you can customize the expander view by using `ExpanderTemplate` property like above example.
+
+## ItemTemplate Selector
+The TreeView allows you to customize the appearance of each item with different templates based on specific constraints by using the `DataTemplateSelector`. You can choose a `DataTemplate` for each item at runtime based on the value of data-bound property using `DataTemplateSelector`.
```
[the remark this answers is not in the document]
Comment (Dhivya-SF4094) [reply MAUI/TreeView/appearance.md:145]: Replaced with the above content

File: MAUI/TreeView/appearance.md [hunk 7]
```diff
@@ -0,0 +1,312 @@
...
+
+### Create a data template selector
+Create custom class that inherits from `DataTemplateSelector`, and override the `OnSelectTemplate` method to return the `DataTemplate` for that item. At runtime, the TreeView invokes the `OnSelectTemplate` method for each item and passes the data object as parameter.
+
+Create different templates and by using `DataTemplateSelector`, load those templates using `OnSelectTemplate` based on requirements.
```
[the remark this answers is not in the document]
Comment (Dhivya-SF4094) [reply MAUI/TreeView/appearance.md:151]: Replaced with the above content

File: MAUI/TreeView/appearance.md [hunk 8]
```diff
@@ -0,0 +1,312 @@
...
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+### Applying a data template selector
+
+Assign custom `DataTemplateSelector` to the `ItemTemplate` either in XAML or C#.
+
+Following code example illustrates to load the different templates for treeview items using `DataTemplateSelector` based on different levels.
```
[the remark this answers is not in the document]
Comment (Dhivya-SF4094) [reply MAUI/TreeView/appearance.md:183]: Replaced with the above content

File: MAUI/TreeView/appearance.md [hunk 9]
```diff
@@ -0,0 +1,312 @@
...
+        else
+            return Template2;
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+### Applying a data template selector
+
+Assign custom `DataTemplateSelector` to the `ItemTemplate` either in XAML or C#.
```
[the remark this answers is not in the document]
Comment (Dhivya-SF4094) [reply MAUI/TreeView/appearance.md:181]: Replaced with the above content

File: MAUI/TreeView/appearance.md [hunk 10]
```diff
@@ -0,0 +1,312 @@
...
+                               ItemTemplate="{StaticResource ItemTemplateSelector}"/>
+    </ContentPage.Content>
+</ContentPage>
+{% endhighlight %}
+{% highlight c# hl_lines="2" %}
+SfTreeView treeView = new SfTreeView();
+treeView.ItemTemplate = new ItemTemplateSelector()
+{% endhighlight %}
+<% endtabs %>
+Download the entire source code from GitHub here
```
[the remark this answers is not in the document]
Comment (Dhivya-SF4094) [reply MAUI/TreeView/appearance.md]: Replaced with the above content

File: MAUI/TreeView/appearance.md [hunk 11]
```diff
@@ -0,0 +1,312 @@
...
+
+Similarly, you can provide `DataTemplateSelector` for `ExpanderTemplate` property.
+
+## Indentation
+
+The TreeView allows customizing the indent spacing of items by setting the `Indentation` property. The default value of this property is `30`. This property can be customized at runtime.
```
[the remark this answers is not in the document]
Comment (Dhivya-SF4094) [reply MAUI/TreeView/appearance.md]: Replaced with the above content

File: MAUI/TreeView/appearance.md [hunk 12]
```diff
@@ -0,0 +1,312 @@
...
+<syncfusion:SfTreeView x:Name="treeView" ExpanderWidth="40">
+{% endhighlight %}
+{% highlight c# hl_lines="2" %}
+SfTreeView treeView = new SfTreeView();
+treeView.ExpanderWidth = 32;
+{% endhighlight %}
+<% endtabs %>
+
+## ExpanderPosition
+The TreeView allows you change the position of expander view by setting the `ExpanderPosition` property. The default value of this property is `Start`.This property has following two positions:
```
[the remark this answers is not in the document]
Comment (Dhivya-SF4094) [reply MAUI/TreeView/appearance.md:242]: Replaced with the above content

File: MAUI/TreeView/appearance.md [hunk 13]
```diff
@@ -0,0 +1,312 @@
...
+{% endhighlight %}
+{% highlight c# hl_lines="2" %}
+SfTreeView treeView = new SfTreeView();
+treeView.ExpanderPosition = ExpanderPosition.End;
+{% endhighlight %}
+<% endtabs %>
+
+## Level based styling
+
+The TreeView allows you to customize the style of `TreeViewItem` based on different levels by using `IValueConverter`.
```
[the remark this answers is not in the document]
Comment (Dhivya-SF4094) [reply MAUI/TreeView/appearance.md:259]: Replaced with the above content

File: MAUI/TreeView/appearance.md [hunk 14]
```diff
@@ -0,0 +1,312 @@
...
+{% endhighlight %}
+{% highlight c# hl_lines="2" %}
+SfTreeView treeView = new SfTreeView();
+treeView.Indentation = 30;
+{% endhighlight %}
+<% endtabs %>
+
+## ExpanderWidth
+
+The TreeView allows customizing the width of expander view by setting the `ExpanderWidth` property. The default value of this property is `32d`. This property can be customized at runtime.
```
[the remark this answers is not in the document]
Comment (Dhivya-SF4094) [reply MAUI/TreeView/appearance.md:228]: Replaced with the above content

File: MAUI/TreeView/scrolling.md [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
+title: Scrolling in .NET MAUI TreeView Control | Syncfusion
+description: Learn here all about Scrolling support in Syncfusion .NET MAUI TreeView (SfTreeView) Control and more.
+platform: MAUI
+control: SfTreeView
+documentation: ug
+---
+
+# Scrolling in .NET MAUI TreeView (SfTreeView)
+
+The .NET MAUI TreeView provides various options to achieve programmatic scrolling. Please walkthrough the below section in detail to achieve the same.
```
[the remark this answers is not in the document]
Comment (Dhivya-SF4094) [reply MAUI/TreeView/scrolling.md]: Replaced with the above content

File: MAUI/TreeView/getting-started.md [hunk 16]
```diff
@@ -84,8 +84,7 @@ namespace GettingStarted
 
 You can create and manage the `TreeViewNode` objects by yourself to display the data in a hierarchical view. To create a tree view, you can use a `TreeView` control and a hierarchy of `TreeViewNode` objects. You can create the node hierarchy by adding one or more root nodes to the TreeView control’s Nodes collection. Each `TreeViewNode` can then have more nodes added to its Children collection. You can nest the tree view nodes to any depth you need.
 
->**Important** 
-`ItemsSource` is an alternative mechanism to `Nodes` for putting content into the TreeView control. You cannot set both `ItemsSource` and `Nodes` at the same time. When you use `ItemsSource`, nodes created for you internally, but you cannot access them from `Nodes` property.
+I> `ItemsSource` is an alternative mechanism to `Nodes` for putting content into the TreeView control. You cannot set both `ItemsSource` and `Nodes` at the same time. When you use `ItemsSource`, nodes created for you internally, but you cannot access them from `Nodes` property.
```
[the remark this answers is not in the document]
Comment (Dhivya-SF4094) [reply MAUI/TreeView/getting-started.md:87]: Replaced with the above content

File: MAUI/TreeView/getting-started.md [hunk 17]
```diff
@@ -633,8 +631,7 @@ public class FileManagerViewModel
 
 You can create a tree view by binding the `ItemsSource` to the hierarchy property descriptors data source. To create a tree view using hierarchical data binding, set a hierarchical collection to the `ItemsSource` property, and then set the `TargetType` and `ChildPropertyName` property values in `HirearchyPropertyDescriptors` .
 
->**Important** 
-`ItemsSource` is an alternative mechanism to `Nodes` for adding content into the TreeView control. You cannot set both `ItemsSource` and `Nodes` at the same time. When you use `ItemsSource`, nodes are created internally, but you cannot access them from the `Nodes` property.
+I> `ItemsSource` is an alternative mechanism to `Nodes` for adding content into the TreeView control. You cannot set both `ItemsSource` and `Nodes` at the same time. When you use `ItemsSource`, nodes are created internally, but you cannot access them from the `Nodes` property.
```
[the remark this answers is not in the document]
Comment (Dhivya-SF4094) [reply MAUI/TreeView/getting-started.md:634]: Replaced with the above content

File: MAUI/TreeView/scrolling.md [hunk 18]
```diff
@@ -0,0 +1,115 @@
...
+
+## Bring Into View
+
+The TreeView allows programmatic scrolling based on the data model and `TreeViewNode` by using the `BringIntoView` method.
```
[the remark this answers is not in the document]
Comment (Dhivya-SF4094) [reply MAUI/TreeView/scrolling.md:16]: Replaced with the above comment

File: MAUI/TreeView/scrolling.md [hunk 19]
```diff
@@ -0,0 +1,115 @@
...
+    var count= viewModel.ImageNodeInfo.Count;
+    var data = viewModel.ImageNodeInfo[count-1];
+    TreeView.BringIntoView(data);
+}
+{% endhighlight %}
+{% endtabs %}
+
+Download the entire source code from GitHub here.
+
+The `BringIntoView` method comprises of other optional parameters to decide on the way in which the child item should come into view.
```
[the remark this answers is not in the document]
Comment (Dhivya-SF4094) [reply MAUI/TreeView/scrolling.md:31]: Replaced with the above content

File: MAUI/TreeView/scrolling.md [hunk 20]
```diff
@@ -0,0 +1,115 @@
...
+
+### Scroll to the child item with animation
+
+The second optional parameter `disableAnimation` in `BringIntoView` method decides whether the scrolling animation should be enabled or disabled when the child item comes into view. By default, the scrolling will be animated.
```
[the remark this answers is not in the document]
Comment (Dhivya-SF4094) [reply MAUI/TreeView/scrolling.md]: Replaced with the above content

File: MAUI/TreeView/scrolling.md [hunk 21]
```diff
@@ -0,0 +1,115 @@
...
+
+
+{% tabs %}
+{% highlight c# hl_line="6" %}
+
+private void BringIntoView_Clicked(object sender, EventArgs e)
+{
+    var count= viewModel.ImageNodeInfo.Count;
+    var data = viewModel.ImageNodeInfo[count-1];
+    // Here, the second optional parameter has been passed as true hence it will disable the animation
```
[the remark this answers is not in the document]
Comment (Dhivya-SF4094) [reply MAUI/TreeView/scrolling.md:47]: Replaced with the above content

File: MAUI/TreeView/scrolling.md [hunk 22]
```diff
@@ -0,0 +1,115 @@
...
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+N> We need to set the `NodePopulationMode` API value as `PopulationMode.Instant` for scrolling to the collapsed item in addition to the additional parameter passed to the `BringIntoView` method.
+
+### Scroll the item into specified position
+
+The fourth optional parameter `scrollToPosition` in `BringIntoView` method allows to position the scrolled item in the view. The scrolled item can take either of the four positions as explained below. The default position is `Start`.
```
[the remark this answers is not in the document]
Comment (Dhivya-SF4094) [reply MAUI/TreeView/scrolling.md:79]: Replaced with the above content

File: MAUI/TreeView/scrolling.md (hunk 1 above)
Comment (Dhivya-SF4094) [reply MAUI/TreeView/scrolling.md:75]: Replaced with the above content

File: MAUI/TreeView/scrolling.md [hunk 23]
```diff
@@ -0,0 +1,115 @@
...
+    // Scrolls to the data item to make visible in the view.
+    treeView.BringIntoView(data, false, false, ScrollToPosition.MakeVisible);
+}
+{% endhighlight %}
+{% endtabs %}
+
+## Scrollbar Visibility
+The TreeView provides an option to enable or disable the `Scrollbar` visibility by using the `IsScrollBarVisible` property. By default, the value will be true.
+
+N> Due to some restrictions in native ScrollView renderer in Xamarin.Forms, you cannot change the `IsScrollBarVisible` value at runtime. It can be defined only when initializing the TreeView.
```
[the remark this answers is not in the document]
Comment (Dhivya-SF4094) [reply MAUI/TreeView/scrolling.md]: Removed this note, as it is not necessary

File: MAUI/TreeView/scrolling.md [hunk 24]
```diff
@@ -0,0 +1,115 @@
...
+
+* `End`: Scroll to make the node positioned at the end of the view.
+
+{% tabs %}
+{% highlight c# hl_line="6"%}
+private void BringIntoView_Clicked(object sender, EventArgs e)
+{
+    var count= viewModel.ImageNodeInfo.Count;
+    var data = viewModel.ImageNodeInfo[count-1];
+    // Scrolls to the data item to make visible in the view.
```
[the remark this answers is not in the document]
Comment (Dhivya-SF4094) [reply MAUI/TreeView/scrolling.md:95]: Replaced with the above content.

**Review by Lakshminatarajan — APPROVED** [review]
Approved

**Review by PiruthivirajM — APPROVED** [review]

**Pull request merged**